crates/biome_js_analyze/src/lint/style/no_parameter_assign.rs

```diff
@@ -64,48 +116,153 @@ declare_lint_rule! {
     }
 }
 
+/// Options for the rule `NoParameterAssign`
+#[derive(Clone, Debug, Deserialize, Deserializable, Eq, PartialEq, Serialize)]
+#[cfg_attr(feature = "schema", derive(schemars::JsonSchema))]
+#[serde(rename_all = "camelCase", deny_unknown_fields, default)]
+pub struct NoParameterAssignOptions {
+    /// Whether to report an error when a dependency is listed in the dependencies array but isn't used. Defaults to `allow`.
+    #[serde(default = "parameter_mutation_default")]
+    pub parameter_mutation: PropertyAssignmentMode,
+}
+
+#[derive(Clone, Debug, Deserialize, Deserializable, Eq, PartialEq, Serialize)]
+#[cfg_attr(feature = "schema", derive(schemars::JsonSchema))]
+#[serde(rename_all = "camelCase", deny_unknown_fields)]
+/// Specifies whether property assignments on function parameters are allowed or denied.
+pub enum PropertyAssignmentMode {
+    /// Allows property assignments on function parameters.
+    /// This is the default behavior, enabling flexibility in parameter usage.
+    Allow,
+
+    /// Disallows property assignments on function parameters.
+    /// Enforces stricter immutability to prevent unintended side effects.
+    Deny,
+}
+
+impl Default for NoParameterAssignOptions {
+    fn default() -> Self {
+        Self {
+            parameter_mutation: parameter_mutation_default(),
+        }
+    }
+}
+
+fn parameter_mutation_default() -> PropertyAssignmentMode {
+    PropertyAssignmentMode::Allow
+}
```

Comment:
```suggestion
#[derive(Clone, Debug, Default, Deserialize, Deserializable, Eq, PartialEq, Serialize)]
#[cfg_attr(feature = "schema", derive(schemars::JsonSchema))]
#[serde(rename_all = "camelCase", deny_unknown_fields)]
/// Specifies whether property assignments on function parameters are allowed or denied.
pub enum PropertyAssignmentMode {
    /// Allows property assignments on function parameters.
    /// This is the default behavior, enabling flexibility in parameter usage.
    #[default]
    Allow,

    /// Disallows property assignments on function parameters.
    /// Enforces stricter immutability to prevent unintended side effects.
    Deny,
}
```